flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/KinesisDataFetcher.java

```diff
@@ -400,20 +482,51 @@ private RecordEmitter createRecordEmitter(Properties configProps) {
 		SequenceNumber lastSequenceNum,
 		MetricGroup metricGroup,
 		KinesisDeserializationSchema<T> shardDeserializer) {
+		return createShardConsumer(
+			subscribedShardStateIndex,
+			subscribedShard,
+			lastSequenceNum,
+			metricGroup,
+			shardDeserializer,
+			null
+		);
+	}
+
+	/**
+	 * Create a new shard consumer.
+	 * Override this method to customize shard consumer behavior in subclasses.
+	 * @param subscribedShardStateIndex the state index of the shard this consumer is subscribed to
+	 * @param subscribedShard the shard this consumer is subscribed to
+	 * @param lastSequenceNum the sequence number in the shard to start consuming
+	 * @param metricGroup the metric group to report metrics to
+	 * @param streamInfoList the stream info used for enhanced fan-out to consume from
+	 * @return shard consumer
+	 */
+	protected ShardConsumer<T> createShardConsumer(
+		Integer subscribedShardStateIndex,
+		StreamShardHandle subscribedShard,
+		SequenceNumber lastSequenceNum,
+		MetricGroup metricGroup,
+		KinesisDeserializationSchema<T> shardDeserializer,
+		@Nullable List<FanOutStreamInfo> streamInfoList) {
```

Comment:
@xiaolong-sn why do we need to pass all the stream info? A `ShardConsumer` is only concerned about a single stream

Reply:
I reverted the change to ShardConsumer.